app/api/auth/create_temp_user.go

```diff
@@ -19,7 +19,7 @@ import (
 // summary: Create a temporary user
 // description: Creates a temporary user and generates an access token valid for 2 hours
 //
-//   * No “Authorization” header should be present
+//   * None of “Authorization” header and "access_token" cookie should be present
```

Comment:
This breaks the scenario I discuss in the previous comment probably.
So: 
- the app use cookies,
- the app switch to tokens
- when starting, as it does not have token, it creates a temp user
- this service fails 

May be pretty hard to debug.

Reply:
What will the app send in the Authorization header when there is no active session and the cookie is disabled? The middleware prefers the Authorization header over the cookie if both are present. So the app should send an empty bearer if it wants the backend to ignore the cookie.

Reply:
Should the backend prefer the Authorization header over the cookie in tempUserCreate & accessTokenCreate/accessTokenRefresh even when use_cookie=0?

Reply:
> What will the app send in the Authorization header when there is no active session and the cookie is disabled (especially in the requests where we do not have the use_cookie parameter)? The middleware prefers the Authorization header over the cookie if both are present. So the app should send a non-empty (but wrong, probably, short to be sure) bearer if it wants the backend to ignore the cookie.

Sending a non-empty wrong bearer when we do not have token ? Wow, very hacky. When the app has no token for a service which returns a token... it just does not send token, which makes sense to me. 

> Should the backend prefer the Authorization header over the cookie in tempUserCreate & accessTokenCreate/accessTokenRefresh even when use_cookie=1?

For this specific service, you could just ignore any authentication instead of failing when there is one given. That looks simple, is there something wrong with that?
For accessTokenCreate, as it requires input (either code/state or token), it is just a matter of priorities I think.

Reply:
So,
1) tempUserCreate & accessTokenCreate/accessTokenRefresh always ignore both the Authorization header and the cookie (no matter what use_cookie's value is) and set the new cookie (if use_cookie=1) without deleting the old one (this may lead to the situation where several different session cookies are set with different attributes).
2) All the services requiring the user to be authenticated prefer the Authorization header over the cookie, so, after logging out (when the UI doesn't use cookies), it is possible that the cookie still exists in the user's browser keeping the use authenticated.
3) authLogout deletes only one session at once (preferring the one from the header), so another session may still remain keeping the user logged in.

Is that really what we want?

Reply:
>     1. tempUserCreate & accessTokenCreate/accessTokenRefresh always ignore both the Authorization header and the cookie (no matter what use_cookie's value is) and set the new cookie (if use_cookie=1) without deleting the old one 

accessTokenCreate/accessTokenRefresh cannot ignore the given token as it may use it to get a new one.  So it can just use in this preference order: code/state, authorization header, cookie

> (this may lead to the situation where several different session cookies are set with different attributes).

So if a browser has two cookies with the same name and different parameters (that it does not send), it may send two cookies with the same name in its "Cookie" header to the backend ? Looks strange.

>     2. All the services requiring the user to be authenticated prefer the Authorization header over the cookie, so, after logging out (when the UI doesn't use cookies), it is possible that the cookie still exists in the user's browser keeping the use authenticated.

The backend would return authenticated-like response while the frontend thinks it is not connected.

>     3. authLogout deletes only one session at once (preferring the one from the header), so another session may still remain keeping the user logged in.

I don't see the difference with the previous one.

> Is that really what we want?

What are the other options?

1. The frontend always provides in header the type of authentication it uses. 

2. When a cookie is provided in addition to an authorization header or a code/state (for tokenCreate), the backend always return a Set-Cookie removing it. But it only works if it is a valid cookie (so it has parameters), otherwise it is unable to do so anyway. 

As a consequence of (2), the logout would logout both token if both an header and cookie would be given.

Does (2) looks good to you ? Otherwise do you have another proposal. Anyway we should re-write the full proposal as it is not straightforward.

Reply:
reworked, should be fine now